STM32MP1 Series support

arnopo opened the pull request:
Add basic support of the Cortex-M4 core for the  STM32MP1 processor.
Provide support for the stm32mp1 DK2 discovery board.

This first step allows to run and debug the Hello_world sample.

arnopo committed:
soc: stm32: add initial soc support for stm32mp1 family

Add the initial SoC support for the STM32MP1 family. The code was
tested on STM32MP157c, but should work on any STM32MP1XX currently
available.

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@st.com>

arnopo committed:
ext: hal: Enable support for stm32mp1xx hal

Provide required changes to stm32cube HAL and LL for the
STM32MP1X series MCU.

Origin: ST Microelectronics
License: BSD-3-Clause
URL: https://github.com/STMicroelectronics/STM32CubeMP1
Commit: 1.1.0
Purpose: HAL layer for STM32MP1
Maintained-by: External

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@st.com>

arnopo committed:
dts: add declaration of stm32mp157

Add the declaration of the ST Microlectronics
stm32mp157 soc.

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@st.com>

arnopo committed:
drivers/interrupt_controller: stm32: add support of stm32mp1

Add support of the stm32mp1 gpio exti.

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@st.com>

galak commented on boards/arm/stm32mp157c_dk2/board.h:
```diff
@@ -0,0 +1,12 @@
+/*
```
board.h shouldn't be needed, please remove it.

galak reviewed (changes requested):
need to remove board.h

dbkinder commented on boards/arm/stm32mp157c_dk2/doc/stm32mp157_disco.rst:
```diff
@@ -0,0 +1,290 @@
+------------
+install `stm32mp1 developer package`_.
+
+1) start OpenOCD in dedicated terminal
```
Lots of formatting issues.  Here's a recommended rewrite:
```
1) start OpenOCD in a dedicated terminal

   - Start up the  sdk environment::

       source <SDK installation directory>/environment-setup-cortexa7hf-neon-vfpv4-openstlinux_weston-linux-gnueabi
 
   - Start OpenOCD::

       ${OECORE_NATIVE_SYSROOT}/usr/bin/openocd -s ${OECORE_NATIVE_SYSROOT}/usr/share/openocd/scripts -f board/stm32mp15x_dk2.cfg

2) run gdb in Zephyr environment

   .. code-block:: console

      # On Linux
      cd $ZEPHYR_BASE/samples/hello_world
      mkdir -p build && cd build

      # Use cmake to configure a Ninja-based build system:
      cmake -GNinja -DBOARD=stm32mp157_dk2 ..

      # Now run ninja on the generated build system:
      ninja debug
```

erwango commented:
recheck

galak reviewed (dismissed):
So I think the resource table we should do as a follow up to this PR.  Since we should figure out how to support for other SoCs as well.

erwango commented on drivers/interrupt_controller/exti_stm32.c:
```diff
@@ -334,12 +501,13 @@ static inline void __stm32_exti_isr_22(void *arg)
 	__stm32_exti_isr(23, 24, arg);
 }
-#endif /* CONFIG_SOC_SERIES_STM32F7X */
+#endif /* CONFIG_SOC_SERIES_STM32{F7X MP1x} */
```
We have sanitycheck script that look for non existing Kconfig symbols and we clean them up.
Remove CONFIG_SOC_SERIES_STM32{F7X MP1x}

erwango reviewed (changes requested):
Initial review

arnop2 commented on ext/hal/st/stm32cube/stm32mp1xx/drivers/include/stm32mp1xx_ll_exti.h:
```diff
@@ -1,6 +1,6 @@
 /**
   ******************************************************************************
-  * @file    stm32wbxx_ll_exti.h
```
i would prefer to keep it separate. This is a fix to apply on top of the ST released(https://github.com/STMicroelectronics/STM32CubeMP1).

erwango replied:
Indeed this should be on top of initial commit, but as an introduction of the file, not as a fix of the file (since stm32mp1_ll_exti.h is not part of official ST release)

arnopo committed:
ext: hal: add stm32mp1 exti ll driver

The stm32mp1 exti ll driver is mandatory for compilation.
add it on top of stm32cubeMP1 version 1.0.0.

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@st.com>

erwango commented on boards/arm/stm32mp157c_dk2/Kconfig.board:
```diff
@@ -0,0 +1,10 @@
+# Kconfig - STM32MP157C Discovery board configuration
+#
+# Copyright (c) 2019 ST Microelectronics Limited
```
Please change copyright to "STMircoelectronics".
"Limited" does not apply.
Apply everywhere

erwango commented on boards/arm/stm32mp157c_dk2/stm32mp157c_dk2_defconfig:
```diff
@@ -0,0 +1,16 @@
+CONFIG_SOC_SERIES_STM32MP1X=y
+CONFIG_SOC_STM32MP15_M4=y
+CONFIG_CORTEX_M_SYSTICK=y
+# 80MHz system clock
```
204?

erwango commented on dts/arm/st/mp1/stm32mp157.dtsi:
```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
```
Resquest is still valid

arnop2 commented on boards/arm/stm32mp157c_dk2/stm32mp157c_dk2_defconfig:
```diff
@@ -0,0 +1,16 @@
+CONFIG_CONSOLE=y
+CONFIG_RAM_CONSOLE=y
+CONFIG_RAM_CONSOLE_BUFFER_SIZE=1024
+CONFIG_SOC_STM32MP15_RPROC_RSC_TABLE=y
```
should depend neither on board or soc but CONFIG:
- CONFIG_RAM_CONSOLE
- CONFIG_OPENAMP
I will try to change it in a more graceful way

erwango commented on soc/arm/st_stm32/stm32mp1/Kconfig.series:
```diff
@@ -0,0 +1,17 @@
+	select HAS_STM32CUBE
+	select CPU_HAS_ARM_MPU
+	select CPU_HAS_SYSTICK
+	select CLOCK_CONTROL_STM32_CUBE if CLOCK_CONTROL
```
Remove this line for now

erwango commented on ext/hal/st/stm32cube/stm32mp1xx/README:
```diff
@@ -0,0 +1,39 @@
+   BSD-3-Clause
+
+License Link:
+   http://www.st.com/resource/en/license_agreement/dm00218346.pdf
```
Still valid

erwango commented on boards/arm/stm32mp157c_dk2/stm32mp157c_dk2.yaml:
```diff
@@ -0,0 +1,21 @@
+arch: arm
+toolchain:
+  - zephyr
+  - gccarmemb
```
- xtools

erwango commented on ext/hal/st/stm32cube/stm32mp1xx/README:
```diff
@@ -36,4 +36,5 @@ License Link:
-    * Initial version
\ No newline at end of file
+    * Initial version 1.0.0
+    * + ext: hal: add stm32mp1 exti ll driver
```
Please add impacted files and internal tracking number if any

erwango reviewed (changes requested):
Few remaining points

erwango commented on ext/hal/st/stm32cube/stm32mp1xx/README:
```diff
@@ -36,4 +36,13 @@ License Link:
+    Impacted files:
+      ext/hal/st/stm32cube/stm32mp1xx/README
+      .../drivers/include/stm32mp1xx_ll_exti.h
+      .../stm32mp1xx/drivers/src/stm32mp1xx_ll_exti.c
```
...drivers/src/stm32mp1xx_ll_exti.c

erwango reviewed (changes requested):
One last comment and we're good to go.

erwango commented on ext/hal/st/stm32cube/stm32mp1xx/drivers/src/stm32mp1xx_ll_exti.c:
```diff
@@ -0,0 +1,327 @@
+/**
+  ******************************************************************************
+  * @file    stm32wbxx_ll_exti.c
```
stm32mp1

galak commented:
Looks good to me, one minor comment is the commit order in the PR seems odd.  The board commit should be the last one.

arnop2 commented:
> Looks good to me, one minor comment is the commit order in the PR seems odd. The board commit should be the last one.

I have changed the commit order and re-push commits.
Thanks

arnop2 commented:
Update done to fix rebase issue on master (compilation failed due to https://github.com/zephyrproject-rtos/zephyr/commit/4da0f8b7965b962e3ac7fc0183d327b275adae18).
Impact in soc/arm/st_stm32/stm32mp1/linker.ld to remove (OPTIONAL)

arnopo committed:
boards: arm: Add support for STM32MP157 Discovery board

Add configuration, pinmux, dts and documentation for the STM32MP157
Discovery board based on the STM32MP157 SoC.

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@st.com>

arnopo committed:
ext: hal: fix naming in rcc ll

The CONFIG_ prefix is reserved in Zephyr, rename CONFIG_SHIFT and
associated constants by adding a RCC_ prefix

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@st.com>

arnop2 commented:
Don't take into account false Kconfig fail: 
reason is that error reports a ext/hal/st/stm32cube/stm32mp1xx/README:51 text which describes the (https://github.com/zephyrproject-rtos/zephyr/pull/13794/commits/9f63849214f76edb154d815cac92df7df1c11923) commit that updateq definition in ext/hal/st/stm32cube/stm32mp1xx/drivers/include/stm32mp1xx_ll_rcc.h to suppress this error check.

arnop2 commented:
Hello Kumar,

Thank for the merge!

Arnaud

From: Kumar Gala <notifications@github.com>
Sent: vendredi 19 avril 2019 19:06
To: zephyrproject-rtos/zephyr <zephyr@noreply.github.com>
Cc: Arnaud POULIQUEN <arnaud.pouliquen@st.com>; Author <author@noreply.github.com>
Subject: Re: [zephyrproject-rtos/zephyr] STM32MP1 Series support (#13794)


Merged #13794<https://github.com/zephyrproject-rtos/zephyr/pull/13794> into master.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/zephyrproject-rtos/zephyr/pull/13794#event-2288627566>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AGCNTA2XHTSIMCANYRNBZPDPRH3WTANCNFSM4G2J5BXQ>.
